### PR TITLE
Document the supports style properties that were deemed stable during 5.6 cycle

### PR DESCRIPTION
**Pull request by oandregal**
This documents how blocks declare support for some style properties -color, font size, spacing, line height- that were deemed stabled during the WordPress 5.6 release cycle.

Note that:

- the properties prepended with `__experimental` are not documented, as they are still in the experimental phase.
- the `color.link` property is no longer marked as experimental but still requires experimental theme support, so this hasn't been documented either.

### CODE REVIEW

**Comment by nosolosw** [comment]
Cross-posting https://github.com/WordPress/gutenberg/issues/24422 for visibility: it'd be great to use the existing block.json data to show which blocks support each property here.

**Comment by nosolosw** [comment]
A prospective text for the block supports dev note:

## Blocks can declare support for style properties

WordPress 5.6 introduces a new API for blocks to declare support for some style properties. When they do, the block editor will automatically show a UI control for the user to set those values.

```js
supports: {
    color: {
        background: true, // Enable background color UI control.
        gradient: true, // Enable gradient color UI control.
        text: true // Enable text color UI control.
    },
    fontSize: true, // Enable font size UI control.
    lineHeight: true, // Enable line height UI control.
    spacing: {
        padding: true // Enable padding UI control.
    }
}
```

### Preset values for UI controls

Some UI controls will present the users with a set of preset values, in addition to being able to set custom ones. In these cases, the presets are taken from the corresponding theme supports:

- [editor-color-palette](https://developer.wordpress.org/block-editor/developers/themes/theme-support/#block-color-palettes): presets for background and text color controls.
- [editor-gradient-presets](https://developer.wordpress.org/block-editor/developers/themes/theme-support/#block-gradient-presets): presets for gradient color control.
- [editor-font-size](https://developer.wordpress.org/block-editor/developers/themes/theme-support/#block-font-sizes): presets for font size control.

### Block attributes and default values

When blocks declare support for a style property, they also gain additional block attributes. These can also be used to set default values:

```js
attributes: {
    backgroundColor: {
        type: 'string',
        default: 'value',
    },
    gradient: {
        type: 'string',
        default: 'value',
    },
    textColor: {
        type: 'string',
        default: 'value',
    },
    fontSize: {
        type: 'string',
        default: 'value'
    },
    style: {
        type: 'Object',
        default: {
            color: {
                background: 'value',
                gradient: 'value',
                link: 'value',
                text: 'value'
            },
            typography: {
                fontSize: 'value',
                lineHeight: 'value',
            },
            spacing: {
                padding: {
                    top: 'value',
                    right: 'value',
                    bottom: 'value',
                    left: 'value'
                }
            }
        }
    }
}
```

The attributes `backgroundColor`, `gradient`, `textColor`, and `fontSize` hold any preset values the user has selected. When the user sets a custom value instead, these are stored in the corresponding key within the `style` attribute.

**Comment by nosolosw** [comment]
@youknowriad feel free to edit directly the comment with [text for the dev note](https://github.com/WordPress/gutenberg/pull/26771#issuecomment-723080772).

File: docs/designers-developers/developers/block-api/block-supports.md [hunk 1]
```diff
@@ -163,3 +309,40 @@ supports: {
     reusable: false
 }
 ```
+
+## spacing
```
Comment (youknowriad) [inline docs/designers-developers/developers/block-api/block-supports.md:313]: Is this stable? I mean the block support is stable but I'm not sure we have a stable way to enable the UI for themes.

**Review by youknowriad — APPROVED** [review]
Great update thanks :)

**Comment by youknowriad** [comment]
@nosolosw the dev note is great, I think we need to clarify the "requirements". Meaning these only apply to the block wrapper and that it's better used in conjunction with `apiVersion: 2`

File: docs/designers-developers/developers/block-api/block-supports.md (hunk 1 above)
Comment (nosolosw) [reply docs/designers-developers/developers/block-api/block-supports.md:313]: It looks like it is, as per https://github.com/WordPress/gutenberg/pull/25788

Comment (nosolosw) [reply docs/designers-developers/developers/block-api/block-supports.md:313]: Added a comment about the things that require theme support to work https://github.com/WordPress/gutenberg/pull/26771/commits/67dd1c5285cb41ab3a95163c7e183f21b9445ecb

**Comment by nosolosw** [comment]
@youknowriad tweaked a bit the dev note adding a longer intro and explaining how it works. How does it look?

File: docs/designers-developers/developers/block-api/block-supports.md (hunk 1 above)
Comment (nosolosw) [reply docs/designers-developers/developers/block-api/block-supports.md:313]: Prepared https://github.com/WordPress/gutenberg/pull/26859 to tweak the docs and signal that `custom-spacing` theme support still requires the plugin to work.

**Comment by youknowriad** [comment]
Awesome. Now that I think about it, it's probably a good idea to expand on the dynamic blocks support for this `get_block_wrapper_attributes`

**Pull request merged**